src/main/java/com/ensao/gi5/lint/visitor/BooleanExpressionVisitor.java

```diff
@@ -0,0 +1,98 @@
+package com.ensao.gi5.lint.visitor;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.ensao.gi5.lint.util.Utils;
+import com.ensao.gi5.lint.wrapper.NameWrapper;
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.expr.ConditionalExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.stmt.DoStmt;
+import com.github.javaparser.ast.stmt.ForStmt;
+import com.github.javaparser.ast.stmt.IfStmt;
+import com.github.javaparser.ast.stmt.ReturnStmt;
+import com.github.javaparser.ast.stmt.WhileStmt;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+
+public class BooleanExpressionVisitor extends VoidVisitorAdapter<List<NameWrapper>> {
+
+	@Override
+	public void visit(IfStmt n, List<NameWrapper> arg) {
+		String name = "The if condition";
+		getNameWrapper(n.getCondition(), name).ifPresent(arg::add);
+
+		super.visit(n, arg);
+	}
+
+	@Override
+	public void visit(ForStmt n, List<NameWrapper> arg) {
+		String name = "The for statement condition";
+		Optional<Expression> compare = n.getCompare();
+		if (!compare.isEmpty()) {
+			getNameWrapper(compare.get(), name).ifPresent(arg::add);
+		}
+
+		super.visit(n, arg);
+	}
+
+	@Override
+	public void visit(DoStmt n, List<NameWrapper> arg) {
+		String name = "The do statement condition";
+		getNameWrapper(n.getCondition(), name).ifPresent(arg::add);
+
+		super.visit(n, arg);
+	}
+
+	@Override
+	public void visit(ConditionalExpr n, List<NameWrapper> arg) {
+		String name = "The ternary operator condition";
+		getNameWrapper(n.getCondition(), name).ifPresent(arg::add);
+
+		super.visit(n, arg);
+	}
+
+	@Override
+	public void visit(WhileStmt n, List<NameWrapper> arg) {
+		String name = "The while statement condition";
+		getNameWrapper(n.getCondition(), name).ifPresent(arg::add);
+
+		super.visit(n, arg);
+	}
+
+	@Override
+	public void visit(ReturnStmt n, List<NameWrapper> arg) {
+		String name = "The return statement condition";
+		Optional<Expression> expression = n.getExpression();
+		if (!expression.isEmpty()) {
+			getNameWrapper(expression.get(), name).ifPresent(arg::add);
+		}
+
+		super.visit(n, arg);
+	}
+
+	private Optional<NameWrapper> getNameWrapper(Expression expr, String name) {
+		if (expr.isBinaryExpr()) {
+			if (checkIfContainsMoreThanTwoOperands(expr)) {
+				int line = Utils.getLine(expr.getBegin());
+				NameWrapper nameWrapper = new NameWrapper(name, line);
+				return Optional.of(nameWrapper);
+			}
+		}
+
+		return Optional.empty();
+	}
+
+	private boolean checkIfContainsMoreThanTwoOperands(Expression expr) {
+		BinaryExpr binaryExpr = (BinaryExpr) expr;
```

Comment:
this cast is not protected.
`ClassCastException` may be raised

Reply:
Thank you, sir, for your comment. As you can see in this context i check at the beginning that the expr is a binary expression before invoking this method. This prevents a `ClassCastException`. So, my question is whether this is the best way to do it or not.

Reply:
Indeed, there is only one usage of this method, which has a prior check if the `expression` is of type `BinaryExpr `,
Imagine when this method is to be used elsewhere, the same check should be done,
That means that this check has to be a part of the method.